tests/test_integration_async.py

```diff
@@ -0,0 +1,617 @@
+
+    @override_settings(ROOT_URLCONF="tests.urls_invalid")
+    async def test_is_toolbar_request_without_djdt_urls(self):
+        """Test cases when the toolbar urls aren't configured."""
+        request = arf.get("/__debug__/render_panel/")
+        self.assertFalse(self.toolbar.is_toolbar_request(request))
+
+        request = arf.get("/render_panel/")
+        self.assertFalse(self.toolbar.is_toolbar_request(request))
+
+    @override_settings(ROOT_URLCONF="tests.urls_invalid")
+    async def test_is_toolbar_request_override_request_urlconf(self):
+        """Test cases when the toolbar URL is configured on the request."""
+        request = arf.get("/__debug__/render_panel/")
+        self.assertFalse(self.toolbar.is_toolbar_request(request))
+
+        # Verify overriding the urlconf on the request is valid.
+        request.urlconf = "tests.urls"
+        self.assertTrue(self.toolbar.is_toolbar_request(request))
+
+    async def test_is_toolbar_request_with_script_prefix(self):
+        """
+        Test cases when Django is running under a path prefix, such as via the
+        FORCE_SCRIPT_NAME setting.
+        """
+        request = arf.get("/__debug__/render_panel/", SCRIPT_NAME="/path/")
+        self.assertTrue(self.toolbar.is_toolbar_request(request))
+
+        request = arf.get("/invalid/__debug__/render_panel/", SCRIPT_NAME="/path/")
+        self.assertFalse(self.toolbar.is_toolbar_request(request))
+
+        request = arf.get("/render_panel/", SCRIPT_NAME="/path/")
+        self.assertFalse(self.toolbar.is_toolbar_request(self.request))
+
+    async def test_data_gone(self):
+        response = await self.async_client.get(
+            "/__debug__/render_panel/?store_id=GONE&panel_id=RequestPanel"
+        )
+        self.assertIn("Please reload the page and retry.", response.json()["content"])
+
+    async def test_sql_page(self):
+        response = await self.async_client.get("/execute_sql/")
+        self.assertEqual(
+            len(response.toolbar.get_panel_by_id("SQLPanel").get_stats()["queries"]), 1
+        )
+
+    async def test_async_sql_page(self):
+        response = await self.async_client.get("/async_execute_sql/")
+        self.assertEqual(
+            len(response.toolbar.get_panel_by_id("SQLPanel").get_stats()["queries"]), 2
+        )
+
+
+# Concurrent database queries are not fully supported by Django's backend with
+# psycopg2 and psycopg3's async support isn't integrated yet. As a result, in
+# ASGI environments(especially with the Django Debug Toolbar), either
+# deadlocks can occure in certain cases like when running
+# tests/views/async_execute_sql_concurrently in ASGI environment
+# or queries will execute synchronously. Check ou
+# https://forum.djangoproject.com/t/are-concurrent-database-queries-in-asgi-a-thing/24136/2
+# https://forum.djangoproject.com/t/are-concurrent-database-queries-in-asgi-a-thing/24136/2
+# https://github.com/django/deps/blob/main/accepted/0009-async.rst#the-orm
```

Comment:
All of that is of course true, but it sounds as if we only looked at PostgreSQL and not at sqlite, mysql etc.

Either mention that the PostgreSQL support is used as an example, or research what happens with other database engines. (The former is fine for me!)